graphene_federation/service.py

```diff
@@ -18,7 +18,7 @@ def _mark_field(
             # todo write tests on regexp
             schema_field_name = to_camel_case(field_name) if auto_camelcase else field_name
             pattern = re.compile(
-                r"(type\s%s\s[^\{]*\{[^\}]*\s%s[\s]*:[\s]*[^\s]+)(\s)" % (
+                r"(type\s%s\s[^\{]*\{[^\}]*\s%s[\s]*(\([^\)]*\))?[\s]*:[\s]*[^\s]+)(\s)" % (
```

Comment:
Allows things like

```python
@extend(fields='id')
class MyObject(graphene.ObjectType):
  id = external(ID(required=True))
  other_field=external(Int(required=True))
  my_filed = require(
    Field(
      Int,
      resolver=my_resolver,
      arg=Int(required=True)
    )
    fields='other_field'  
  )
```

which resolve (without the directive) to something like this:

```
extend type MyObject @key(fileds="id") {
  id: Int! @external
  other_field: Int! @external
  
  my_field(arg: Int): Int
}
```

to be augmented correctly to:

```
extend type MyObject @key(fileds="id") {
  id: Int! @external
  other_field: Int! @external
  
  my_field(arg: Int): Int @require(field="other_field")
}
```